The RenderView layer shouldn't need to set shouldRasterize for backdrop filters.

mattwoodrow opened the pull request:
#### 26c33a981702dd6c1df6d4d0da255b68290d47f8
<pre>
The RenderView layer shouldn&apos;t need to set shouldRasterize for backdrop filters.
<a href="https://bugs.webkit.org/show_bug.cgi?id=273611">https://bugs.webkit.org/show_bug.cgi?id=273611</a>
&lt;<a href="https://rdar.apple.com/127364620">rdar://127364620</a>&gt;

Reviewed by Simon Fraser.

Doing so uses more memory, and shouldn&apos;t be needed if the RenderView is opaque.

Skip making the RenderView a backdrop root for the GraphicsLayers if its opaque.
Make sure any backdrop filters that will capture up to the root can still use the
opaque blur settings.

* LayoutTests/css3/filters/backdrop/resource-use-add-more-layers-expected.txt:
* LayoutTests/css3/filters/backdrop/resource-use-excessive-expected.txt:
* LayoutTests/css3/filters/backdrop/resource-use-ok-expected.txt:
* LayoutTests/platform/glib/css3/filters/backdrop/resource-use-add-more-layers-expected.txt:
* LayoutTests/platform/glib/css3/filters/backdrop/resource-use-excessive-expected.txt:
* LayoutTests/platform/glib/css3/filters/backdrop/resource-use-ok-expected.txt:
* LayoutTests/platform/mac-wk1/css3/filters/backdrop/resource-use-add-more-layers-expected.txt: Copied from LayoutTests/css3/filters/backdrop/resource-use-add-more-layers-expected.txt.
* LayoutTests/platform/mac-wk1/css3/filters/backdrop/resource-use-excessive-expected.txt: Copied from LayoutTests/css3/filters/backdrop/resource-use-excessive-expected.txt.
* LayoutTests/platform/mac-wk1/css3/filters/backdrop/resource-use-ok-expected.txt: Copied from LayoutTests/css3/filters/backdrop/resource-use-ok-expected.txt.
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::dumpProperties const):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::flushCompositingState):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateBackdropRoot):

Canonical link: <a href="https://commits.webkit.org/278397@main">https://commits.webkit.org/278397@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04b386ad41186b91e2937b0c90bcfe7f21f2cb7c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50390 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29684 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2692 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53649 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1080 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52693 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35926 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41102 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52489 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27346 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43380 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22206 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24761 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/637 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8768 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/46740 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/699 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55241 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25488 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/622 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48511 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26749 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43555 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47549 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11055 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27611 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26481 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (approved):
Can we test this via a layer tree dump?

mattwoodrow committed:
The RenderView layer shouldn't need to set shouldRasterize for backdrop filters.
https://bugs.webkit.org/show_bug.cgi?id=273611
<rdar://127364620>

Reviewed by Simon Fraser.

Doing so uses more memory, and shouldn't be needed if the RenderView is opaque.

Skip making the RenderView a backdrop root for the GraphicsLayers if its opaque.
Make sure any backdrop filters that will capture up to the root can still use the
opaque blur settings.

* LayoutTests/css3/filters/backdrop/resource-use-add-more-layers-expected.txt:
* LayoutTests/css3/filters/backdrop/resource-use-excessive-expected.txt:
* LayoutTests/css3/filters/backdrop/resource-use-ok-expected.txt:
* LayoutTests/platform/glib/css3/filters/backdrop/resource-use-add-more-layers-expected.txt:
* LayoutTests/platform/glib/css3/filters/backdrop/resource-use-excessive-expected.txt:
* LayoutTests/platform/glib/css3/filters/backdrop/resource-use-ok-expected.txt:
* LayoutTests/platform/mac-wk1/css3/filters/backdrop/resource-use-add-more-layers-expected.txt: Copied from LayoutTests/css3/filters/backdrop/resource-use-add-more-layers-expected.txt.
* LayoutTests/platform/mac-wk1/css3/filters/backdrop/resource-use-excessive-expected.txt: Copied from LayoutTests/css3/filters/backdrop/resource-use-excessive-expected.txt.
* LayoutTests/platform/mac-wk1/css3/filters/backdrop/resource-use-ok-expected.txt: Copied from LayoutTests/css3/filters/backdrop/resource-use-ok-expected.txt.
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::dumpProperties const):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::flushCompositingState):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateBackdropRoot):

Canonical link: https://commits.webkit.org/278397@main
